src/pages/index.js

```diff
@@ -32,30 +92,86 @@ const cardList = new Section(
   cardListSelector
 );
 
-cardList.renderItems(initialCards);
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__avatar"
+);
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    // take  user data from api and render to profile section
+    userInfo.setUserInfo(userData);
+    userInfo.setAvatar(userData);
+    // take initial cards from api and render to card section
+    cardList.renderItems(cards.reverse());
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const editAvatarModal = new PopupWithForm("#profile-edit-avatar", (link) => {
+  editAvatarModal.renderLoading(true);
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      editAvatarModal.renderLoading(false);
```

Comment:
Needs correction: Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

The same for other requests

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally